include/experimental/__p1673_bits/blas1_linalg_add.hpp

```diff
@@ -97,13 +129,23 @@ void add(in_object_1_t x,
 }
 
 template<class ExecutionPolicy,
-         class in_object_1_t,
-         class in_object_2_t,
-         class out_object_t>
-void add(ExecutionPolicy&& /* exec */,
-         in_object_1_t x,
-         in_object_2_t y,
-         out_object_t z)
+         class ElementType_x,
+         class Extents_x,
+         class Layout_x,
+         class Accessor_x,
+         class ElementType_y,
+         class Extents_y,
+         class Layout_y,
+         class Accessor_y,
+         class ElementType_z,
+         class Extents_z,
+         class Layout_z,
+         class Accessor_z>
+void add(
+  ExecutionPolicy&& /* exec */,
+  std::experimental::basic_mdspan<ElementType_x, Extents_x, Layout_x, Accessor_x> x,
```

Comment:
I'll need to think a bit how to constrain (not mandate -- mandating would be easier) x, y, and z to all have the same rank, but this is perfectly fine for now -- thank you! : - D

Reply:
I think we could do it like this:

1. Make the extents for each of x, y, and z a `ptrdiff_t...` parameter pack (e.g., `ptrdiff_t... Extents_x`)
2. Add a `requires` clause (establishing a constraint) after the template parameters: `requires sizeof...(Extents_x) == sizeof...(Extents_y) && sizeof...(Extents_x) == sizeof...(Extents_y)`

Daisy is crazy good at parameter pack magic.  I don't think we need it to be too fancy here.  Anyway, we'll need to make several passes to catch all the constraints, so it's totally up to you if we want to do this now or later.  Thanks! : - D

Reply:
I'll just do it now so I get better at constraints :-)

Thanks for the suggestion!